scene/resources/particle_process_material.cpp

```diff
@@ -984,6 +980,10 @@ void ParticleProcessMaterial::_update_shader() {
 		}
 		code += "		\n";
 		code += "		vec3 noise_direction = get_noise_direction(TRANSFORM[3].xyz);\n";
+		// set the noise direction to up so the turbulence color is "neutral"
+		code += "		if (COLLIDED && length(VELOCITY) <= 3.0) {\n";
```

Comment:
This will hurt performance, see:
* https://github.com/godotengine/godot/pull/83749

Reply:
Thanks! I will rebase first and check it.

Reply:
Also small style nitpick, please remember to capitalize the first letter and end with a period on comments.